### PR TITLE
Parse as float and then round to int

### LINKED ISSUE #355: RESOLUTION as float raises exception
The `RESOLUTION` value in `EXT-X-STREAM-INF` is listed as being a decimal value so some outputs list it with `.0` and that causes the parsing to fail ( see https://datatracker.ietf.org/doc/html/rfc8216#section-4.3.4.2 )

### PR DESCRIPTION
**Pull request by daveisfera**
Fixes #355

### CODE REVIEW

File: m3u8/model.py [hunk 1]
```diff
@@ -959,7 +959,10 @@ def __init__(self, uri, stream_info, media, base_uri):
         if resolution is not None:
             resolution = resolution.strip('"')
             values = resolution.split("x")
-            resolution_pair = (int(values[0]), int(values[1]))
+            resolution_pair = (
+                int(round(float(values[0]))),
```
Comment (leandromoreira) [inline m3u8/model.py:963]: just out of curiosity: is there any packager outputting manifests with decimal resolution?

**Review by leandromoreira — APPROVED** [review]

File: m3u8/model.py (hunk 1 above)
Comment (daveisfera) [reply m3u8/model.py:963]: I had some example files I was testing with that had `1280.0x720.0` but I'm not sure where they came from. `ffmpeg` [outputs them as integers](https://github.com/FFmpeg/FFmpeg/blob/e38092ef9395d7049f871ef4d5411eb410e283e0/libavformat/hlsplaylist.c#L86) and I assume that most do as well

Comment (leandromoreira) [reply m3u8/model.py:963]: no worries, thank you

**Comment by leandromoreira** [comment]
H264 uses integers to represent resolution. `pic_width_in_mbs_minus_1 ue(v)`
image

**Comment by daveisfera** [comment]
Closing since this is out of spec

**Pull request closed without merging**